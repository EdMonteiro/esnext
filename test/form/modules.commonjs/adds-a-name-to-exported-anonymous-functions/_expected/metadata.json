{
  "modules.commonjs": {
    "imports": [],
    "exports": [
      {
        "type": "named-export",
        "bindings": [
          {
            "exportName": "add",
            "localName": "add"
          }
        ],
        "node": {
          "type": "ExpressionStatement",
          "expression": {
            "type": "AssignmentExpression",
            "operator": "=",
            "left": {
              "type": "MemberExpression",
              "object": {
                "type": "Identifier",
                "name": "exports",
                "typeAnnotation": null
              },
              "property": {
                "type": "Identifier",
                "name": "add",
                "typeAnnotation": null
              },
              "computed": false
            },
            "right": {
              "type": "FunctionExpression",
              "id": null,
              "params": [
                {
                  "type": "Identifier",
                  "name": "a",
                  "typeAnnotation": null
                },
                {
                  "type": "Identifier",
                  "name": "b",
                  "typeAnnotation": null
                }
              ],
              "body": {
                "type": "BlockStatement",
                "directives": [],
                "body": [
                  {
                    "type": "ReturnStatement",
                    "argument": {
                      "type": "BinaryExpression",
                      "operator": "+",
                      "left": {
                        "type": "Identifier",
                        "name": "a",
                        "typeAnnotation": null
                      },
                      "right": {
                        "type": "Identifier",
                        "name": "b",
                        "typeAnnotation": null
                      }
                    }
                  }
                ]
              },
              "generator": false,
              "async": false,
              "returnType": null,
              "typeParameters": null
            }
          }
        }
      }
    ],
    "directives": []
  }
}