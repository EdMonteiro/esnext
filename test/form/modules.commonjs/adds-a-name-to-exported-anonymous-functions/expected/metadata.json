{
  "modules.commonjs": {
    "imports": [],
    "exports": [
      {
        "bindings": [
          {
            "exportName": "add",
            "localName": "add"
          }
        ],
        "node": {
          "type": "ExpressionStatement",
          "expression": {
            "type": "AssignmentExpression",
            "left": {
              "type": "MemberExpression",
              "computed": false,
              "object": {
                "type": "Identifier",
                "name": "exports"
              },
              "property": {
                "type": "Identifier",
                "name": "add"
              }
            },
            "operator": "=",
            "right": {
              "type": "FunctionExpression",
              "body": {
                "type": "BlockStatement",
                "body": [
                  {
                    "type": "ReturnStatement",
                    "argument": {
                      "type": "BinaryExpression",
                      "operator": "+",
                      "left": {
                        "type": "Identifier",
                        "name": "a"
                      },
                      "right": {
                        "type": "Identifier",
                        "name": "b"
                      }
                    }
                  }
                ]
              },
              "expression": false,
              "generator": false,
              "id": null,
              "params": [
                {
                  "type": "Identifier",
                  "name": "a"
                },
                {
                  "type": "Identifier",
                  "name": "b"
                }
              ]
            }
          }
        }
      }
    ],
    "directives": []
  }
}